hazelcast/invocation.py

```diff
@@ -265,17 +192,13 @@ def _send(self, invocation, connection, ignore_heartbeat):
         try:
             connection.send_message(message)
         except IOError as e:
+            self._listener_service.remove_event_handler(correlation_id)
```

Comment:
a check like `if invocation.event_handler is not None` as you did above may be better to not do the removal for non-listener related invocations